test/threadsafe_function/threadsafe_function.cc

```diff
@@ -0,0 +1,85 @@
+#define NAPI_EXPERIMENTAL
+#include <uv.h>
+#include "napi.h"
+
+using namespace Napi;
+
+const size_t ARRAY_LENGTH = 10;
+const size_t MAX_QUEUE_SIZE = 2;
+
+// Thread data to transmit to JS
+static int ints[ARRAY_LENGTH];
+
+static uv_thread_t uv_threads[2];
+static napi_threadsafe_function ts_fn;
+static ThreadSafeFunction ts_fn_wrap;
+
+int sum = 0;
+
+static void data_source_thread(void* data) {
+  int args = 15;
+  ts_fn_wrap.Call([args](Napi::Env env, Napi::Function jsCallback) {
+    jsCallback.Call({ Number::New(env, args) });
+  }, napi_tsfn_blocking);
+  sum += 5;
+  if (data) {}
```

Comment:
Warnings about unused parameters should be prevented by doing
```C++
(void) data;
```
at the top of the function.